Library/Paginator.swift

```diff
@@ -0,0 +1,136 @@
+import Combine
+import Foundation
+
+/**
+ Used to coordinate the process of paginating through values. This class is specific to the type of pagination
+ in which a page's results contains a cursor that can be used to request the next page of values.
+
+ This class is designed to work with SwiftUI/Combine. For an example, see `PaginationExampleView.swift`.
+
+ This class is generic over the following types:
+
+ * `Value`:         The type of value that is being paginated, i.e. a single row, not the array of rows. The
+                    value must be equatable.
+ * `Envelope`:      The type of response we get from fetching a new page of values.
+ * `SomeError`: The type of error we might get from fetching a new page of values.
+ * `Cursor`:        The type of value that can be extracted from `Envelope` to request the next page of
+                    values.
+ * `RequestParams`: The type that allows us to make a request for values without a cursor.
+
+ - parameter valuesFromEnvelope:   A function to get an array of values from the results envelope.
+ - parameter cursorFromEnvelope:   A function to get the cursor for the next page from a results envelope.
+ - parameter requestFromParams:    A function to get a request for values from a params value.
+ - parameter requestFromCursor:    A function to get a request for values from a cursor value.
+
+  You can observe the results of `values`, `isLoading`, `error` and `state` to access the loaded data.
+
+ */
```

Comment:
Awesome documentation!

Reply:
I take very little credit, this is largely copy-pasted from the original `paginate` 🤫